pgspecial/dbcommands.py

```diff
@@ -28,36 +30,36 @@
 
 @special_command("\\l", "\\l[+] [pattern]", "List databases.", aliases=("\\list",))
 def list_databases(cur, pattern, verbose):
-    query = '''SELECT d.datname as "Name",
+    query = SQL('''SELECT d.datname as "Name",
         pg_catalog.pg_get_userbyid(d.datdba) as "Owner",
         pg_catalog.pg_encoding_to_char(d.encoding) as "Encoding",
         d.datcollate as "Collate",
         d.datctype as "Ctype",
-        pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges"'''
+        pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges"''')
     if verbose:
-        query += ''',
+        query += SQL(''',
             CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT')
                     THEN pg_catalog.pg_size_pretty(pg_catalog.pg_database_size(d.datname))
                     ELSE 'No Access'
             END as "Size",
             t.spcname as "Tablespace",
-            pg_catalog.shobj_description(d.oid, 'pg_database') as "Description"'''
-    query += """
+            pg_catalog.shobj_description(d.oid, 'pg_database') as "Description"''')
+    query += SQL("""
     FROM pg_catalog.pg_database d
-    """
+    """)
     if verbose:
-        query += """
+        query += SQL("""
         JOIN pg_catalog.pg_tablespace t on d.dattablespace = t.oid
-        """
+        """)
     params = {}
     if pattern:
-        query += """
+        query += SQL("""
         WHERE d.datname ~ %(schema_pattern)s
-        """
+        """)
         _, schema = sql_name_pattern(pattern)
-        params["schema_pattern"] = schema
-    query = query + " ORDER BY 1"
-    cur.execute(query, params)
+        params["schema_pattern"] = Literal(schema)
+    query = query + SQL(" ORDER BY 1")
+    cur.execute(query.format(**params))
```

Comment:
@dvarrazzo This statement fails because `.format()` is only defined in the SQL class. But the result of the `__add__()` method is of the parent class type `Composable`. I'm not really sure how to circumvent this.

Reply:
In this specific case you can attach the pattern to the WHERE bit:
```python
query += SQL("WHERE d.datname ~ {}").format(schema)
```
should do (schema is wrapped automatically in a `Literal`).

More in general, you can create the main skeleton of the query and inject the variable parts as parameter, instead of chaining a sequence. In this case it should be (untested):

```python
    params = {}
    query = SQL('''
        SELECT d.datname as "Name",
        pg_catalog.pg_get_userbyid(d.datdba) as "Owner",
        pg_catalog.pg_encoding_to_char(d.encoding) as "Encoding",
        d.datcollate as "Collate",
        d.datctype as "Ctype",
        pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges
        {verbose_fields}
        FROM pg_catalog.pg_database d
        {verbose_tables}
        {pattern_where}
        ORDER BY 1
        "''')

    if verbose:
        params["verbose_fields"] = SQL(''',
            CASE WHEN pg_catalog.has_database_privilege(d.datname, 'CONNECT')
                    THEN pg_catalog.pg_size_pretty(pg_catalog.pg_database_size(d.datname))
                    ELSE 'No Access'
            END as "Size",
            t.spcname as "Tablespace",
            pg_catalog.shobj_description(d.oid, 'pg_database') as "Description"''')
        params["verbose_tables"] = SQL("""
            JOIN pg_catalog.pg_tablespace t on d.dattablespace = t.oid
            """)
    else:
        params["verbose_fields"] = SQL("")
        params["verbose_tables"] = SQL("")

    if pattern:
        _, schema = sql_name_pattern(pattern)
        params["schema_pattern"] = schema

        params["pattern_where"] = SQL("""
        WHERE d.datname ~ {}
        """).format(schema)
    else:
        params["pattern_where"] = SQL("")

    cur.execute(query.format(**params))
```

Reply:
Thank you! That makes sense. I've updated the PR with the "correct" implementation for this function.

I'll update the rest of the commands later today.